### PR TITLE
Add ObjectTests

### PR DESCRIPTION
**Pull request by hallucinogen**
Add around 75% tests to the public APIs of `ParseObject`

Will add the rest (that includes async functions) in next PR.

### CODE REVIEW

**Comment by hallucinogen** [comment]
Fun fact, https://ci.appveyor.com/project/hallucinogen/parse-sdk-dotnet/build/1.0.238 said the commit was made 3 months ago. LOL

File: Parse/Public/ParseObject.cs [hunk 1]
```diff
@@ -428,8 +428,8 @@ string propertyName
     /// <param name="traverseParseObjects">Whether to traverse into ParseObjects' children</param>
     /// <param name="yieldRoot">Whether to include the root in the result</param>
     /// <returns></returns>
-    internal static IEnumerable<object> DeepTraversal(
-        object root, bool traverseParseObjects = false, bool yieldRoot = false) {
+    internal static IEnumerable<object> DeepTraversal(object root,
+        bool traverseParseObjects = false, bool yieldRoot = false) {
```
Comment (richardjrossiii) [inline Parse/Public/ParseObject.cs]: Necessary changes? I think we can either put these in another PR or just discard them, no?

Comment (hallucinogen) [inline Parse/Public/ParseObject.cs]: I'll put them in another PR for formatting
